qcodes_contrib_drivers/drivers/Keysight/Keysight_E36313A.py

```diff
@@ -0,0 +1,111 @@
+from qcodes import VisaInstrument
+from qcodes import Instrument
+from qcodes.instrument.channel import InstrumentChannel
+
+
+class E36313AChannel(InstrumentChannel):
+    """
+
+    """
+    def __init__(self, parent: Instrument, name: str, chan: int) -> None:
+        """
+        Args:
+            parent (Instrument): The instrument to which the channel is
+            attached.
+            name (str): The name of the channel
+            channum (int): The number of the channel in question (1-2)
+        """
+        # Sanity Check inputs
+        if name not in ['ch1', 'ch2', 'ch3']:
+            raise ValueError("Invalid Channel: {}, expected 'ch1' or 'ch2' or 'ch3'"
+                             .format(name))
+        if chan not in [1, 2, 3]:
+            raise ValueError("Invalid Channel: {}, expected '1' or '2' or '3'"
+                             .format(chan))
+
+        super().__init__(parent, name)
+
```

Comment:
the parameters need to be reviewed wrt when they can be snapshotted (inspiration can be taken from https://github.com/qcodes/Qcodes/blob/master/qcodes/instrument_drivers/yokogawa/GS200.py#L481)

Reply:
Thanks for pointing this out. This was in my mind too. I have advised Lukas that first he tries to make all parameters correct wrt to SCPI commands, then check if their behaviour is expected and finally look which snapshot fails. Those failing snapshots can be because of different modes of the instrument and inspiration for fixing that can be taken from Yokogawa, as you have pointed out.